### PR TITLE
Remove deactivation and GracefulSwitchLb from MultiChildLb

### PR DESCRIPTION
**Pull request by ejona86**
I can leave the three commits as separate when merging. The changes to MultiChildLb itself are pretty easy. Most of the review will be in the two earlier commits which remove the dependency on these features.

I think this exposes some other potential cleanups, but I didn't want to bring in too much.

### CODE REVIEW

File: util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java [hunk 1]
```diff
@@ -357,12 +335,12 @@ public final ChildLbState getChildLbStateEag(EquivalentAddressGroup eag) {
   }
 
   /**
-   * Filters out non-ready and deactivated child load balancers (subchannels).
+   * Filters out non-ready child load balancers (subchannels).
    */
   protected final List<ChildLbState> getReadyChildren() {
     List<ChildLbState> activeChildren = new ArrayList<>();
     for (ChildLbState child : getChildLbStates()) {
-      if (!child.isDeactivated() && child.getCurrentState() == READY) {
```
Comment (larry-safran) [inline util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java:365]: I don't see how deactivated children in ClusterManagerLoadBalancer get excluded from the ready children and are not actively used.

Comment (ejona86) [reply util/src/main/java/io/grpc/util/MultiChildLoadBalancer.java:365]: `ClusterManagerLoadBalancer.acceptResolvedAddresses()` injects deactivated children back into the config, and deactivates them when it first notices they are removed. When it notices they are added back to the config, it reactivates them. That was really the only substantive change for ClusterManagerLb, as it moved that logic from MultiChildLb to ClusterManagerLb.

So both active and deactive children are passed down to MultiChildLB. `updateOverallBalancingState()` was already ignoring deactivated children for ConnectivityState calculations and the picker.

File: xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java [hunk 2]
```diff
@@ -101,23 +102,32 @@ protected Map<Object, ChildLbState> createChildLbMap(ResolvedAddresses resolvedA
...
+      ClusterManagerConfig config = (ClusterManagerConfig)
+          resolvedAddresses.getLoadBalancingPolicyConfig();
+      ClusterManagerConfig lastConfig = (ClusterManagerConfig)
+          lastResolvedAddresses.getLoadBalancingPolicyConfig();
+      Map<String, PolicySelection> adjChildPolicies = new HashMap<>(config.childPolicies);
+      for (Entry<String, PolicySelection> entry : lastConfig.childPolicies.entrySet()) {
+        ClusterManagerLbState state = (ClusterManagerLbState) getChildLbState(entry.getKey());
+        if (adjChildPolicies.containsKey(entry.getKey())) {
+          if (state.deletionTimer != null) {
+            state.reactivateChild();
```
Comment (larry-safran) [inline xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java:116]: How does `getChildLbState` return a deactivated child when `MultiChildLoadBalancer.getRemovedChildren` now does `ChildLbState childLbState = childLbStates.remove(key);`?

File: xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java [hunk 3]
```diff
@@ -101,23 +102,32 @@ protected Map<Object, ChildLbState> createChildLbMap(ResolvedAddresses resolvedA
...
-
-      return acceptRetVal.status;
-    } finally {
-      resolvingAddresses = false;
+      config = new ClusterManagerConfig(adjChildPolicies);
+      resolvedAddresses =
+          resolvedAddresses.toBuilder().setLoadBalancingPolicyConfig(config).build();
     }
+    lastResolvedAddresses = resolvedAddresses;
+    return super.acceptResolvedAddresses(resolvedAddresses);
```
Comment (larry-safran) [inline xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java:130]: What keeps this from calling `shutdown` on all of the removed addresses in MultiChildLoadBalancer.java:127 `      shutdownRemoved(acceptRetVal.removedChildren);`

File: xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java (hunk 2 above)
Comment (ejona86) [reply xds/src/main/java/io/grpc/xds/ClusterManagerLoadBalancer.java:116]: It isn't removed while deactivated. Here are the important snippets:

```java
// for removed policies, we add them back
          adjChildPolicies.put(entry.getKey(), entry.getValue());
// and then repackage the config
      config = new ClusterManagerConfig(adjChildPolicies);
      resolvedAddresses =
          resolvedAddresses.toBuilder().setLoadBalancingPolicyConfig(config).build();
```

The model is ClusterManagerLb tells MultiChildLb _all_ the children it is interested in. And MultiChildLb manages those. ClusterManagerLb managers the list of children it is interested in.

**Review by larry-safran — APPROVED** [review]

**Pull request merged**